src/metabase/models/permissions.clj

```diff
@@ -776,7 +829,65 @@
 
 ;;; ----------------------------------------------- Graph Updating Fns -----------------------------------------------
 
-(s/defn ^:private update-table-read-perms!
+(defn- download-permissions-set
+  [group-id]
+  (map :object
+       (db/select [Permissions :object]
+                  {:where [:and
+                           [:= :group_id group-id]
+                           [:or
+                            [:= :object (hx/literal "/")]
+                            [:like :object (hx/literal "/download/%")]]]})))
```

Comment:
what you have here is a list, not a set.

Or you can use `db/select-field`, it returns a set by default.
```suggestion
  (db/select-field :object [Permissions :object]
                   {:where [:and
                            [:= :group_id group-id]
                            [:or
                             [:= :object (hx/literal "/")]
                             [:like :object (hx/literal "/download/%")]]]}))
```

Reply:
Good point, the code works the same for any kind of collection, but I like your revised query